File: src/backend/MyAIRunningMate/MyAIRunningMate.Contracts/Analytics/Responses/UserMetricsResponse.cs
namespace MyAIRunningMate.Contracts.Analytics.Responses;

public record UserMetricsResponse(
    double WeightKg,
    double UserVolumetricOxygenMax,
    int UserMaxHeartRate,
    int UserLactateThresholdHeartRate,
    int UserLactateThresholdPower,
    double UserLactateThresholdSpeed,
    string UserVolumetricOxygenMaxRating,
    int FitnessPercentile,
    double PowerToWeightRatio,
    string PowerRating,
    double ThresholdPercentagePower,
    string TrainingLevel,
    string FitnessRankColor
);

File: src/frontend/MyAIRunningMate-UI/src/types/dashboard/userMetricsResponse.ts
export interface  UserMetricsResponse {
    last_recorded_time: string;
    age: number;
    weight_kg: number;
    user_volumetric_oxygen_max: number;
    user_max_heart_rate: number;
    user_lactate_threshold_heart_rate: number;
    user_lactate_threshold_power: number;
    user_lactate_threshold_speed: number;
    user_volumetric_oxygen_max_rating: string;
    fitness_percentile: number;
    power_to_weight_ratio: number;
    power_rating: string;
    threshold_percentage_power: number;
    training_level: string;
    fitness_rank_color: string;
}